src/main/java/com/cloudbees/jenkins/support/slowrequest/SlowRequestChecker.java

```diff
@@ -122,11 +131,30 @@ protected void doRun() throws Exception {
                 }
             }
         }
+
+        if (slowRequestCount >= MINIMAL_SLOW_REQUEST_COUNT) {
+            boolean newThreadDumps = SlowRequestThreadDumpsGenerator.checkThreadDumpsTrigger(iota);
+
+            if (newThreadDumps) {
+                try {
+                    SlowRequestThreadDumpsGenerator slowRequestThreadDumpsGenerator =
+                            new SlowRequestThreadDumpsGenerator(iota);
+                    slowRequestThreadDumpsGenerator.start();
```

Comment:
We are blocking the `SlowRequestChecker` here for the entire time that the `SlowRequestThreadDumpsGenerator` shall run. Which is by default 5s * 4, so 20s. Way over the 3s recurrence period of the checker. Maybe we need to launch this asynchronously or something..